jupyterlab_celltests/lint.py

```diff
@@ -7,71 +7,52 @@
-    passed = True
-
-    if lines_per_cell:
-        for i, lines_in_cell in enumerate(metadata.get('cell_lines', [])):
-            if lines_in_cell <= lines_per_cell:
-                ret.append(LintMessage(i+1, 'Checking lines in cell', LintType.LINES_PER_CELL, True))
-            else:
-                ret.append(LintMessage(i+1, 'Checking lines in cell', LintType.LINES_PER_CELL, False))
-                passed = False
-    return ret, passed
+    if lines_per_cell < 0:
+        return [], True
+    for i, lines_in_cell in enumerate(metadata.get('cell_lines', [])):
+        # TODO: ambiguous - e.g. cell 0 or first cell?
+        ret.append(LintMessage(i+1, 'Checking lines in cell', LintType.LINES_PER_CELL, lines_in_cell <= lines_per_cell))
+    return ret, all([x.passed for x in ret])
 
 
 def lint_cells_per_notebook(cells_per_notebook, metadata):
-    ret = []
-    passed = True
-
-    if cells_per_notebook:
-        if metadata.get('cell_count', -1) <= cells_per_notebook:
-            ret.append(LintMessage(-1, 'Checking cells per notebook', LintType.CELLS_PER_NOTEBOOK, True))
-        else:
-            ret.append(LintMessage(-1, 'Checking cells per notebook', LintType.CELLS_PER_NOTEBOOK, False))
-            passed = False
-    return ret, passed
+    if cells_per_notebook < 0:
+        return [], True
+    passed = metadata.get('cell_count', -1) <= cells_per_notebook
+    return [LintMessage(-1, 'Checking cells per notebook', LintType.CELLS_PER_NOTEBOOK, passed)], passed
 
 
 def lint_function_definitions(function_definitions, metadata):
-    ret = []
-    passed = True
-
-    if function_definitions:
-        if metadata.get('functions', -1) <= function_definitions:
-            ret.append(LintMessage(-1, 'Checking functions per notebook', LintType.FUNCTION_DEFINITIONS, True))
-        else:
-            ret.append(LintMessage(-1, 'Checking functions per notebook', LintType.FUNCTION_DEFINITIONS, False))
-            passed = False
-    return ret, passed
+    if function_definitions < 0:
+        return [], True
+    passed = metadata.get('functions', -1) <= function_definitions
+    return [LintMessage(-1, 'Checking functions per notebook', LintType.FUNCTION_DEFINITIONS, passed)], passed
 
 
 def lint_class_definitions(class_definitions, metadata):
-    ret = []
-    passed = True
-
-    if class_definitions:
-        if metadata.get('classes', -1) <= class_definitions:
-            ret.append(LintMessage(-1, 'Checking classes per notebook', LintType.CLASS_DEFINITIONS, True))
-        else:
-            ret.append(LintMessage(-1, 'Checking classes per notebook', LintType.CLASS_DEFINITIONS, False))
-            passed = False
-    return ret, passed
+    if class_definitions < 0:
+        return [], True
+    passed = metadata.get('classes', -1) <= class_definitions
+    return [LintMessage(-1, 'Checking classes per notebook', LintType.FUNCTION_DEFINITIONS, passed)], passed
 
 
+# TODO: I think this isn't lint and should be removed.
```

Comment:
I'd argue this is indeed a lint (but maybe of a different kind).

Reply:
Isn't it just "test coverage", which people would usually expect to see named as such (rather than being under lint)?